### PR TITLE
Doc-174 Conditional and branch filtering

### PR DESCRIPTION
**Pull request by karensawrey**
This addresses the request to add more information on how branch filter pattern, step-level branch filtering, and conditionals in pipelines all interact with each other.

### CODE REVIEW

File: pages/pipelines/conditionals.md.erb [hunk 1]
```diff
@@ -6,20 +6,22 @@ You can define conditionals at the step level in your `pipeline.yml` or at the p
...
+You can have complete control over when to trigger pipeline builds by using conditional expressions to filter incoming webhooks. You need to define conditionals in the _Pipeline Settings page_ for your repository provider to run builds only when expressions evaluate to `true`. For example, to run only when a branch is tagged `Approved`:
 
 <%= image "conditionals.png", width: 864, height: 298, alt: "Conditional Filtering settings" %>
 
-Conditionals are supported in [Bitbucket](/docs/integrations/bitbucket), [Bitbucket Server](/docs/integrations/bitbucket-server), [GitHub](/docs/integrations/github), [GitHub Enterprise](/docs/integrations/github-enterprise), and [GitLab](/docs/integrations/gitlab) (including GitLab Community and GitLab Enterprise).
+Limiting at the pipeline level affects whether builds get created at all, so conditional filtering has the highest priority. Build conditionals are run before any other build qualifier tests, so branch filters always run second. If both are present, such build will have to pass both filters to be created – first the pipeline-level limiting filter and then the conditional filter.
```
Comment (ticky) [inline pages/pipelines/conditionals.md.erb:15]: Some of this gets into a bit of jargon which we probably don't need to share;

```suggestion
Pipeline-level build conditionals are evaluated before any other build trigger settings. If both a conditional and a branch filter are present, both filters must pass for a build to be created – first the pipeline-level limiting filter and then the conditional filter.
```

**Review by ticky — APPROVED** [review]
I've made some suggestions, but this looks good!

**Comment by karensawrey** [comment]
Thanks a million for reviewing this PR, @ticky! Highly appreciated. I applied all the suggested changes - they made the explanations much clearer!

@plaindocs, your move.

File: pages/pipelines/branch_configuration.md.erb [hunk 2]
```diff
@@ -2,21 +2,23 @@
...
 
-In step-level and pipeline-level branch filtering you can use `*` as a wildcard, and `!` for not, as shown in the [examples](#branch-pattern-examples). If you want a full range of regular expressions that operate on more than branch names, take a look at the [conditionals](/docs/pipelines/conditionals) page.
+In step-level and pipeline-level branch filtering, you can use `*` as a wildcard, and `!` for not, as shown in the [examples](#branch-pattern-examples). If you want a full range of regular expressions that operate on more than branch names, take a look at the [conditionals](/docs/pipelines/conditionals) page.
 
 {:toc}
 
 ## Pipeline-level branch filtering
 
-By default a pipeline will trigger builds for all branches (`*` or blank). In your pipeline settings you can set specific branch patterns for the entire pipeline. If a commit doesn't match the branch pattern, no build will be created.
+By default, a pipeline will trigger builds for all branches (`*` or blank). In your pipeline settings, you can set specific branch patterns for the entire pipeline. If a commit doesn't match the branch pattern, no build will be created.
```
Comment (plaindocs) [inline pages/pipelines/branch_configuration.md.erb:11]:
```suggestion
By default, a pipeline triggers builds for all branches (`*` or blank). In your pipeline settings, you can set specific branch patterns for the entire pipeline. If a commit doesn't match the branch pattern, no build is created.
```

File: pages/pipelines/branch_configuration.md.erb [hunk 3]
```diff
@@ -2,21 +2,23 @@
...
 <%= image "pipeline-level.png", width: 864, height: 306, alt: "Pipeline-level branch filtering" %>
 
-<section class="Docs__troubleshooting-note">
-  <h3>Pull Request Builds</h3>
-  <p>Builds created for pull requests will ignore any pipeline-level branch filters. If you want to limit the branches that can build pull requests, add an additional branch filter in your pipeline's source control settings.</p>
-  <p>This filter can be found under 'Build pull requests' if you have chosen the 'Trigger builds after pushing code' option.</p>
-</section>
+## Additional branch filtering for pull request builds
+
+Builds created for pull requests will ignore any pipeline-level branch filters. If you want to limit the branches that can build pull requests, add an additional branch filter in your pipeline's source control settings.
```
Comment (plaindocs) [inline pages/pipelines/branch_configuration.md.erb:17]:
```suggestion
Builds created for pull requests ignore any pipeline-level branch filters. If you want to limit the branches that can build pull requests, add an additional branch filter in your pipeline's source control settings.
```

File: pages/pipelines/branch_configuration.md.erb [hunk 4]
```diff
@@ -2,21 +2,23 @@
...
+
+This filter can be found under 'Build pull requests' if you have chosen the 'Trigger builds after pushing code' option.
```
Comment (plaindocs) [inline pages/pipelines/branch_configuration.md.erb:19]:
```suggestion
Find this filter can be found under _Build pull requests_ if you have chosen the _Trigger builds after pushing code_ option.
```

File: pages/pipelines/conditionals.md.erb [hunk 5]
```diff
@@ -6,20 +6,22 @@ You can define conditionals at the step level in your `pipeline.yml` or at the p
...
+You can have complete control over when to trigger pipeline builds by using conditional expressions to filter incoming webhooks. You need to define conditionals in the _Pipeline Settings page_ for your repository provider to run builds only when expressions evaluate to `true`. For example, to run only when a branch is tagged `Approved`:
 
 <%= image "conditionals.png", width: 864, height: 298, alt: "Conditional Filtering settings" %>
 
-Conditionals are supported in [Bitbucket](/docs/integrations/bitbucket), [Bitbucket Server](/docs/integrations/bitbucket-server), [GitHub](/docs/integrations/github), [GitHub Enterprise](/docs/integrations/github-enterprise), and [GitLab](/docs/integrations/gitlab) (including GitLab Community and GitLab Enterprise).
+Pipeline-level build conditionals are evaluated before any other build trigger settings. If both a conditional and a branch filter are present, both filters must pass for a build to be created – first the pipeline-level limiting filter and then the conditional filter.
```
Comment (plaindocs) [inline pages/pipelines/conditionals.md.erb:15]: :+1:

**Review by plaindocs — APPROVED** [review]
:+1:

File: pages/pipelines/branch_configuration.md.erb (hunk 4 above)
Comment (plaindocs) [reply pages/pipelines/branch_configuration.md.erb:19]: I think these are UI elements, but if options as we discussed you might need to revert

Comment (karensawrey) [reply pages/pipelines/branch_configuration.md.erb:19]: I'd say these are options.
Options

Comment (karensawrey) [reply pages/pipelines/branch_configuration.md.erb:19]: I rewrote the beginning of the sentence as suggested, but not touching the 's.

**Pull request merged**